core-kotlin-modules/core-kotlin-lang-oop/src/main/kotlin/com/baeldung/companionobject/CompanionObject.kt

```diff
@@ -0,0 +1,62 @@
+package com.baeldung.companionobject
+
+class CompanionObject {
+    class ClassName {
+        fun anotherFun() {
+            println(propertyName)
+        }
+
+        companion object {
+            const val propertyName: String = "Something..."
+            fun funName() {
+                //...
+            }
+        }
+    }
+
+    class MyClass {
+        companion object Factory {
+            fun createInstance(): MyClass = MyClass()
+        }
+    }
+
+
```

Comment:
Let's add the interface with a companion object here too:

```suggestion
    class MyClass {
        companion object Factory {
            fun createInstance(): MyClass = MyClass()
        }
    }

    interface MyInterface { 
        companion object { 
            const val PROPERTY = "value"
        }
    }

```